File: Cano-Rodriguez-Claudio/Procesos/C/process.c
#include <stdio.h>
#include <unistd.h>
#include <stdlib.h>
#include <time.h>

#define MAX 5 // Countdown limit

int main() {
    struct timespec ts;
    ts.tv_sec = 0;
    ts.tv_nsec = 500000000; // 500 milliseconds

    for (int i = 1; i <= 3; i++) { // Create 3 child processes
        pid_t pid = fork();

        if (pid == -1) {
            fprintf(stderr, "Error: Failed to create child process\n");
            return 1;
        }

        if (pid == 0) { // Child process
            for (int k = MAX; k >= 1; k--) {
                printf("Child Process [%d]: Countdown = %d\n", getpid(), k);
                nanosleep(&ts, NULL);
            }
            exit(0);
        }
    }

    return 0; // Parent process exits
}

File: Cano-Rodriguez-Claudio/Procesos/Python/process.py
import os
import time

MAX = 5  # Countdown limit

for i in range(3):  # Create 3 child processes
    pid = os.fork()

    if pid == 0:  # Child process
        for k in range(MAX, 0, -1):
            print(f"Child Process [{os.getpid()}]: Countdown = {k}")
            time.sleep(0.5)  # Sleep for 500ms
        os._exit(0)  # Terminate child process

# Parent process exits
